metadata-ingestion/src/datahub/ingestion/api/registry.py

```diff
@@ -30,6 +32,27 @@ def is_enabled(self, key: str) -> bool:
         tp = self._mapping[key]
         return not isinstance(tp, Exception)
 
+    def load(self, path: str, cls: type) -> None:
+        for entry_point in pkg_resources.iter_entry_points(path):
+            name = entry_point.name
+            module = None
+
+            try:
+                module = entry_point.load()
+            except ImportError as e:
+                self.register_disabled(name, e)
+
+            # Get all non-abstract classes from module
+            plugin_classes = inspect.getmembers(
```

Comment:
If we use the `attr` syntax from entrypoints, we should be able to get rid the `getmembers` call. The subclass and isabstract checks are probably worth keeping to ensure people don't get weird runtime errors.